backend/apid/middlewares/authorization_test.go

```diff
@@ -2,92 +2,100 @@ package middlewares
 
 import (
 	"context"
+	"errors"
 	"net/http"
+	"net/http/httptest"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	jwt "github.com/dgrijalva/jwt-go"
 	sensujwt "github.com/sensu/sensu-go/backend/authentication/jwt"
-	"github.com/sensu/sensu-go/backend/authorization"
-	"github.com/sensu/sensu-go/testing/mockstore"
 	"github.com/sensu/sensu-go/types"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 )
 
-type TestHandler struct {
-	reqCtx context.Context
-}
-
-type TestResponseWriter struct {
-	header http.Header
+type FakeAuthorizer struct {
+	authorized bool
+	err        error
 }
 
-func (w TestResponseWriter) Header() http.Header {
-	if w.header == nil {
-		w.header = http.Header{}
+func (a *FakeAuthorizer) Authorize(reqInfo *types.RequestInfo) (bool, error) {
+	if a.err != nil {
+		return false, a.err
 	}
-	return w.header
-}
-
-func (w TestResponseWriter) Write(data []byte) (int, error) {
-	return 0, nil
-}
-
-func (w TestResponseWriter) WriteHeader(status int) {}
-
-func (h *TestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	h.reqCtx = r.Context()
+	return a.authorized, nil
 }
 
 func TestAuthorization(t *testing.T) {
-	assert := assert.New(t)
-
-	user := &types.User{
-		Username: "sensu",
-		Password: "passw0rd",
-		Roles:    []string{"canhazreadaccess"},
-		Disabled: false,
-	}
-
-	claims := types.Claims{
-		StandardClaims: jwt.StandardClaims{
-			Subject: user.Username,
+	cases := []struct {
+		name            string
+		claims          *types.Claims
+		reqInfo         *types.RequestInfo
+		authorizedError error
+		isAuthorized    bool
+		expected        int
+	}{
+		{
+			name:     "missing JWT",
```

Comment:
Maybe rename this to "missing claims and request info" (or something in that vein). When I first read "missing JWT" I assumed it was a test where the request itself is missing an Authorization header and it didn't make sense to me that it would return an internal error.